api/app/resources/bookings/booking/booking_post.py

```diff
@@ -0,0 +1,47 @@
+'''Copyright 2018 Province of British Columbia
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.'''
+
+from flask_restplus import Resource
+from flask import request
+from app.schemas.bookings import BookingSchema
+from qsystem import api, api_call_with_retry, db, oidc
+
+
+@api.route("/bookings/", methods=["POST"])
+class BookingPost(Resource):
+
+    booking_schema = BookingSchema()
+
+    @oidc.accept_token(require_token=True)
+    @api_call_with_retry
+    def post(self):
+
+        json_data = request.get_json()
+
+        if not json_data:
+            return {"message": "No input data received for creating a booking"}, 400
+
+        booking, errors = self.booking_schema.load(json_data)
+
+        if errors:
+            print(errors)
+            return {"message": errors}, 422
+
+        db.session.add(booking)
```

Comment:
We should probably be checking that the foreign keys for exam and room both match the office of the CSR here